In [2]:
print("Welcome to my statistics notes, I hope they'll help you")

Welcome to my statistics notes, I hope they'll help you


### Population and Sample
Before starting other stuff, we really need to understand the meaning of population and sample. Suppose, you have selected 10 students at random from your school and measured their weight. These 10 people represent all the students in your school, which makes them a sample of size 10, and all the students at your school are the popualation.

Imagine we have some ungrouped data about the weights of all MacDonald's burger variations. There are 10 bugers and we have their weights in grams. Their weights are,
$$150, 120, 230, 70, 110, 90, 92, 130, 122, 120$$
You need to find -
* The mean
* The mode
* The median
* The standard deviation

### Mean
Mean is really important, and all it means is the average. How do you find the mean? You add all the samples and simply divide them with the sample size. But there are other ways to do it too.
The formula for sample mean is,
$$\overline{x} = \frac{\sum{xi}}{n}$$

In [1]:
import numpy as np
data = [150,120,230,70,110,90,92,130,122,120]
sum = np.sum(data)

In [8]:
mean = sum/10

In [9]:
mean

123.4

So, the mean is 123.4
Now let's sort the data first. Which will make the it easier to answer the next questions.

In [11]:
data.sort()

In [12]:
data

[70, 90, 92, 110, 120, 120, 122, 130, 150, 230]

Here the ***Mode*** is the most freqent value. The only repeated value here is 120, so it is our mode. But mode can be several values in a dataset.<br/>
The ***Median*** is the middle value of the sorted data. Here we can see the sample size $n$ is even, so no middle value. So we take the two value in the middle, $\frac{n}{2}$ and $\frac{n}{2} + 1$ and find their average. Which is 120 in this case.

### Variance and Standard Deviation
Variance is the average of how far the observations are from the mean of a dataset. Standard Deviation is the square root of the variance. It is the measure of spread of the data. When we see the formula for calculating this it will be much more clearer. The sample variance is -
$$s^2 = \frac{\sum{(x_i-\bar{x})^2}}{n-1}$$
The sample standard deviation is,
$$s = \sqrt{\frac{\sum{(x_i-\bar{x})^2}}{n-1}}$$
As we can see, for every $x_i$ value we are finding it's difference with our $\bar{x}$(**mean**) and then we find the average of it. Square root of it gives us the standard deviation

Let's calculate the standard deviation of our above example.

In [7]:
import math
from math import sqrt
def variance(data, mean, n):
    total = 0
    for x_i in data:
        total += (x_i - mean)**2
    return total/(n - 1)

def stdev(data, mean, n):
    return sqrt(variance(data, mean, n))

In [17]:
variance(data, mean, 10)

1919.1555555555558

In [21]:
s = stdev(data, mean, 10)
s

43.8081676808738

In [22]:
s == sqrt(1919.1555555555558)

True

### Personal Workings
These are my personal workings from problem sets of my school. Ignore these as most of these are not gonna make any sense.

In [23]:
data = [94,110,70,115,122,99,86,103,80,91]
n = 10

In [24]:
sum = np.sum(data)

In [28]:
mean = sum/n
mean

97.0

In [27]:
stdev(data, mean, n)

16.131404843417148

In [14]:
import math
from math import sqrt
data = [638, 845, 649, 658, 798]
random = 0
for x in data:
    random += (x-717.6)**2
sqrt(random/(5-1))

96.552058496958

In [9]:
stdev(data, 5, 717.6)

59.95955237271883

In [11]:
96.55/sqrt(5)

43.178472645520934

In [12]:
0.8944-0.0668

0.8276

In [13]:
0.8276*250

206.9

In [20]:
total = 0
data = [638, 845, 649, 658, 798]
for x in data:
    total += (x-717.6)**2
sqrt(total/4)

96.552058496958